mcs/class/System.Web/System.Web.Hosting/HostingEnvironment.cs

```diff
@@ -146,6 +151,16 @@ public override object InitializeLifetimeService ()
 
 		public static void InitiateShutdown ()
 		{
+			shutdownCancellationSource.Cancel ();
+
+			Task[] tasks;
+			lock (runningBackgroundTasks) {
+				tasks = runningBackgroundTasks.ToArray ();
+			}
+			if (runningBackgroundTasks.Length > 0) {
```

Comment:
Ah, no, I meant `tasks`. Guess that'll teach me to do last minute edits w/o compiling.